Botany reagents Tweak

Fox-McCloud opened the pull request:
Said I'd do this once the botany PR was merged and things appeared stable.

Adjusts most of the botany produce to have plantmatter as opposed to "Nutriment". Composting will work with plant matter as well as nutriment, as will the biogenerator.

Gives Blumpkins plasma to retain an existing botany feedback loop (glowshrooms+blumpkins = unstable mutagen).

Also re-adds fish water; oversight for that being axed.

Fixes pod grown diona not being allied with plants and vines.

:cl: Fox McCloud
tweak: adjusts most botany plant reagents to have "plantmatter" as opposed to nutriment
tweak: Blumpkins now have plasma in them
fix: Fixes plant grown diona not being allied with plants and vines
/:cl:

FalseIncarnate commented on code/modules/hydroponics/hydroponics.dm:
```diff
@@ -521,6 +521,12 @@
 
+
+	//Fish Water is both an excellent fertilizer and waters
+	if(S.has_reagent("fishwater", 1))
```
This is an absolutely horrendous if chain. Hilarious, considering how much you complained about this same type of thing in the past.

Fox-McCloud replied:
It's not great, no; both botany systems had this type of horrendous if chain to handle reagents, though there's far more reagents, with this system that interact with plants. I disliked it greatly, but I wasn't about to do a refactor of a refactor amidst 230-some file changes.

Crazylemon64 replied:
Wasn't the table approach old botany used a pretty good approach for handling many generic chems easily, without a big old nasty if chain

Fox-McCloud replied:
The old botany used a big `switch` and repeated `if` statements. It just wasn't as large because it didn't have as many reagents: https://github.com/ParadiseSS13/Paradise/blob/f1c3a7cd13510ecfd99137835ad6ffbe3ffe86de/code/modules/hydroponics/trays/tray_reagents.dm#L56

Fox-McCloud commented on code/modules/hydroponics/grown/ambrosia.dm:
```diff
@@ -80,7 +80,7 @@
 	potency = 10
 	mutatelist = list()
-	reagents_add = list("thc" = 0.15, "kelotane" = 0.15, "bicaridine" = 0.1, "bath_salts" = 0.20, "nutriment" = 0.05)
+	reagents_add = list("thc" = 0.15, "kelotane" = 0.15, "bicaridine" = 0.1, "bath_salts" = 0.20, "plantmatter" = 0.05)
```
The gaia, mimana, and holy melons I intentional kept as nutriment, otherwise, it's virtually impossible to acquire; anything of a more ethereal/magical/supernatural nature I let have nutriment, since it's a universal nutrition provider.

Fox-McCloud commented on code/modules/hydroponics/grown/flowers.dm:
```diff
@@ -15,7 +15,7 @@
 	icon_dead = "poppy-dead"
 	mutatelist = list(/obj/item/seeds/poppy/geranium, /obj/item/seeds/poppy/lily)
-	reagents_add = list("bicaridine" = 0.2, "nutriment" = 0.05)
+	reagents_add = list("bicaridine" = 0.2, "plantmatter" = 0.05)
```
Because those are horrifically toxic, while ingested; these are meant to provide medicinal benefit when consumed or utilized---not poison you horrendously when you eat a "healthy" salad.

FalseIncarnate replied:
Could replace them with something safer like sal glu or another safer existing chem, point was it seemed strange to re-add the chems

Fox-McCloud commented on code/modules/hydroponics/grown/banana.dm:
```diff
@@ -92,7 +92,7 @@
 	mutatelist = list()
 	genes = list(/datum/plant_gene/trait/slip, /datum/plant_gene/trait/teleport, /datum/plant_gene/trait/repeated_harvest)
-	reagents_add = list("singulo" = 0.2, "banana" = 0.1, "vitamin" = 0.04, "nutriment" = 0.02)
+	reagents_add = list("singulo" = 0.2, "banana" = 0.1, "vitamin" = 0.04, "plantmatter" = 0.02)
```
See above comment.

Fox-McCloud commented on code/modules/hydroponics/grown/melon.dm:
```diff
@@ -12,7 +12,7 @@
 	genes = list(/datum/plant_gene/trait/repeated_harvest)
 	mutatelist = list(/obj/item/seeds/watermelon/holy)
-	reagents_add = list("water" = 0.2, "vitamin" = 0.04, "nutriment" = 0.2)
+	reagents_add = list("water" = 0.2, "vitamin" = 0.04, "plantmatter" = 0.2)
```
See above comment.

Fox-McCloud commented:
Addressed most of the reagents/blood tomato/comfrey/aloe things.

A few plants intentional have nutriment over plant matter so it's something botanists can actually shoot for, if they want---again, most plants that were magical/ethereal/supernatural in nature, I let have nutriment as opposed to plant matter.

Fox-McCloud commented on code/modules/hydroponics/grown/cereals.dm:
```diff
@@ -11,7 +11,7 @@
 	icon_dead = "wheat-dead"
 	mutatelist = list(/obj/item/seeds/wheat/oat, /obj/item/seeds/wheat/meat)
-	reagents_add = list("nutriment" = 0.04)
+	reagents_add = list("plantmatter" = 0.04)
```
Addressed.

Fox-McCloud commented:
added plasma to Blumpkins.

On TG botany if you have blumpkins and glowshrooms you can produce mutagen (as their mutagen recipe is radium+phosphorus+chlorine). Ours is a bit different, but this functionality should remain in; as such, I've added plasma to Blumpkins to retain the same pattern.
